backend/src/middleware/validation/validationMiddleware.spec.js

```diff
@@ -0,0 +1,244 @@
+      describe('null', () => {
+        it('throws UserInputError', async () => {
+          createPostVariables = { ...createPostVariables, categoryIds: null }
+          await expect(
+            mutate({ mutation: createPostMutation, variables: createPostVariables }),
+          ).resolves.toMatchObject({
+            data: { CreatePost: null },
+            errors: [
+              {
+                message: 'You cannot save a post without at least one category or more than three',
+              },
+            ],
+          })
+        })
+      })
+
+      describe('empty', () => {
+        it('throws UserInputError', async () => {
+          createPostVariables = { ...createPostVariables, categoryIds: [] }
+          await expect(
+            mutate({ mutation: createPostMutation, variables: createPostVariables }),
+          ).resolves.toMatchObject({
+            data: { CreatePost: null },
+            errors: [
+              {
+                message: 'You cannot save a post without at least one category or more than three',
+              },
+            ],
+          })
+        })
+      })
+
+      describe('more than 3 categoryIds', () => {
+        it('throws UserInputError', async () => {
+          createPostVariables = {
+            ...createPostVariables,
+            categoryIds: ['cat9', 'cat27', 'cat15', 'cat4'],
+          }
+          await expect(
+            mutate({ mutation: createPostMutation, variables: createPostVariables }),
+          ).resolves.toMatchObject({
+            data: { CreatePost: null },
+            errors: [
+              {
+                message: 'You cannot save a post without at least one category or more than three',
+              },
+            ],
+          })
+        })
+      })
+    })
+  })
+
+  describe('validateUpdatePost', () => {
+    describe('post created without categories somehow', () => {
+      let owner, updatePostVariables
+      beforeEach(async () => {
+        const postSomehowCreated = await neode.create('Post', {
+          id: 'how-was-this-created',
+        })
+        owner = await neode.create('User', {
+          id: 'author-of-post-without-category',
+          slug: 'hacker',
+        })
+        await postSomehowCreated.relateTo(owner, 'author')
+        authenticatedUser = await owner.toJson()
+        updatePostVariables = {
+          id: 'how-was-this-created',
+          title: 'I am  a title',
+          content: 'Some content',
+          categoryIds: [],
+        }
+      })
+
+      it('requires at least one category for successful update', async () => {
+        await expect(
+          mutate({ mutation: updatePostMutation, variables: updatePostVariables }),
+        ).resolves.toMatchObject({
+          data: { UpdatePost: null },
+          errors: [
+            { message: 'You cannot save a post without at least one category or more than three' },
+          ],
+        })
+      })
+    })
+  })
+})
```

Comment:
I did not add test to `validateReport` since I've done so in another branch, and the implementation for `reporting` has changed so much it wouldn't make it worth it